src/main/java/com/minecolonies/coremod/generation/defaults/DefaultEnchanterCraftingProvider.java

```diff
@@ -0,0 +1,403 @@
+                .item(enchantedBook(Enchantments.BANE_OF_ARTHROPODS, 5), 1)
+                .item(enchantedBook(Enchantments.BLAST_PROTECTION, 5), 1)
+                .item(enchantedBook(Enchantments.DEPTH_STRIDER, 5), 1)
+                .item(enchantedBook(Enchantments.FEATHER_FALLING, 5), 1)
+                .item(enchantedBook(Enchantments.FIRE_ASPECT, 5), 1)
+                .item(enchantedBook(Enchantments.FIRE_PROTECTION, 5), 1)
+                .item(enchantedBook(Enchantments.FLAME, 5), 1)
+                .item(enchantedBook(Enchantments.FROST_WALKER, 5), 1)
+                .item(enchantedBook(Enchantments.INFINITY, 1), 1)
+                .item(enchantedBook(Enchantments.KNOCKBACK, 5), 1)
+                .item(enchantedBook(Enchantments.LOOTING, 5), 1)
+                .item(enchantedBook(Enchantments.MENDING, 1), 1)
+                .item(enchantedBook(Enchantments.POWER, 5), 1)
+                .item(enchantedBook(Enchantments.PROJECTILE_PROTECTION, 5), 1)
+                .item(enchantedBook(Enchantments.PROTECTION, 5), 1)
+                .item(enchantedBook(Enchantments.PUNCH, 5), 1)
+                .item(enchantedBook(Enchantments.RESPIRATION, 5), 1)
+                .item(enchantedBook(Enchantments.SHARPNESS, 5), 1)
+                .item(enchantedBook(Enchantments.SMITE, 5), 1)
+                .item(enchantedBook(Enchantments.SWEEPING, 5), 1)
+                .item(enchantedBook(Enchantments.UNBREAKING, 5), 1)
+                .item(enchantedBook(ModEnchants.raiderDamage, 2), 1)
+                .build());
+
+        recipeProvider = new EnchanterRecipeProvider(generatorIn);
+        lootTableProvider = new EnchanterLootTableProvider(generatorIn);
+    }
+
+    @NotNull
+    private ItemStack enchantedBook(@NotNull final Enchantment enchantment, final int level)
+    {
+        final ItemStack stack = new ItemStack(Items.ENCHANTED_BOOK);
+        EnchantedBookItem.addEnchantment(stack, new EnchantmentData(enchantment, level));
+        return stack;
+    }
+
+    @NotNull
+    @Override
+    public String getName()
+    {
+        return "EnchanterCraftingProvider";
+    }
+
+    @Override
+    public void act(@NotNull final DirectoryCache cache) throws IOException
+    {
+        recipeProvider.act(cache);
+        lootTableProvider.act(cache);
+    }
+
+    private static class EnchanterRecipeProvider extends CustomRecipeProvider
+    {
+        public EnchanterRecipeProvider(@NotNull final DataGenerator generatorIn)
+        {
+            super(generatorIn);
+        }
+
+        @Override
+        protected void registerRecipes(@NotNull final Consumer<IFinishedRecipe> consumer)
+        {
+            final CompoundNBT tag = new CompoundNBT();
+            tag.putBoolean(NbtTagConstants.TAG_IGNORE_NBT, true);
+
+            final List<ItemStack> tome = Collections.singletonList(new ItemStack(ModItems.ancientTome, 1));
+            tome.get(0).setTag(tag);

```

Comment:
It was necessary to ignore NBT in the recipe since the Ancient Tome has three different states (never handled by a player vs. last handled on a day when there was or wasn't a raid) which are considered incompatible by default.

I suspect (given some other reports on Discord) that the research system might have a similar issue with this item (although that might have only been on an older version).  (And perhaps we might want to use this tag on some other recipes -- IIRC there was a similar compatibility issue with Quark and clocks/compasses.)

Reply:
Yeah, there were a couple inconsistencies due to this issue on research for the More Scrolls, has been fixed.  Research now assumes that anything without explicit NBT data can accept any NBT state, and ItemStorage can handle that, even if there's not much in InventoryUtils to help with it.

The Quark problem goes deep into Structurize, and unfortunately it's also an issue with dyeable armor and banners, and there's not such an easy separation there between cases where we don't care about NBT and those we do.